### ISSUE TITLE
Suggest replacing str::split with empty pattern with str::chars

### ISSUE THREAD

**Issue by Ravenslofty** [rust-lang/rust-clippy#12311]
### What it does

I was helping a friend learn Rust by going through Advent of Code 2020, and they needed to split a `&str` into a `Vec<char>`. They found the `str::split` method, and wrote `string.split("").collect::<Vec<char>>()`. While reviewing their code, this tripped me up a bit, since I would think of `string.chars().collect::<Vec<char>>()` instead. I think this would be a good candidate for a lint.

### Advantage

- I think this is more intuitive than splitting with an empty pattern

### Drawbacks

- as [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=989ce40e85e9edac3867febd6aef006e) shows, this is not a one-to-one replacement because `str::split("")` has leading and trailing empty strings.

### Example

```rust
"hello world".split(""); 
```
The above should trigger a suggestion of replacing with
```
"hello world".chars()
```
